torch/_decomp/decompositions.py

```diff
@@ -4211,6 +4211,28 @@ def floor_divide(self, other):
     return torch.div(self, other, rounding_mode="floor")
 
 
+@register_decomposition(aten.sum.default)
+def sum_default(
+    self: Tensor,
+    keepdim: bool = False,
+    *,
+    dtype: Optional[torch.dtype] = None,
+    out: Optional[Tensor] = None,
+) -> Tensor:
+    if out is None:
+        return aten.sum.dim_IntList(self, [], keepdim=keepdim, dtype=dtype)
+    else:
+        return aten.sum.IntList_out(self, [], keepdim=keepdim, dtype=dtype, out=out)
+
+
+@register_decomposition([aten.squeeze.default, aten.squeeze.dim])
+def squeeze_default(self: Tensor, dim: Optional[int] = None):
+    if dim is None:
+        return aten.squeeze.dims(self, list(range(self.dim())))
+    else:
+        return aten.squeeze.dims(self, [dim])
```

Comment:
Isn't this case already handled in the reference for squeeze?

Reply:
The reference implementation for squeeze calls `prims.squeeze`; for core decompositions, we want to avoid decomposing into prim ops (i.e. only decompose to other ATen operators). Hence why a new decomp was added instead of using the reference impl directly.